src/api/java/com/minecolonies/api/colony/requestsystem/requester/IRequester.java

```diff
@@ -0,0 +1,30 @@
+package com.minecolonies.api.colony.requestsystem.requester;
+
+import com.minecolonies.api.colony.requestsystem.location.ILocation;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Interface that describes an object that can be located in the Minecraft universe and can request objects inside a colony.
+ */
+public interface IRequester
```

Comment:
This class is not supposed to be in this PR yet.
Just the Factories part and the IToken part (and there respective implementations)